python/ray/autoscaler/_private/aws/config.py

```diff
@@ -107,89 +100,105 @@ def log_to_cli(config):
     cli_logger.doassert(provider_name is not None,
                         "Could not find a pretty name for the AWS provider.")
 
+    head_node_type = config["head_node_type"]
+    head_node_config = config["available_node_types"][head_node_type][
+        "node_config"]
+
     with cli_logger.group("{} config", provider_name):
 
         def same_everywhere(key):
-            return config["head_node"][key] == config["worker_nodes"][key]
+            prev_node_config = None
+            for node_type in config["available_node_types"].values():
+                node_config = node_type["node_config"]
+                if prev_node_config is not None:
+                    curr_value = node_config.get(key)
+                    prev_value = prev_node_config.get(key)
+                    if prev_value != curr_value:
+                        return False
+                prev_node_config = node_config
+            return True
```

Comment:
node_config should never be None,
Can we make a python set in which we put all the keys and assert it has 1 value instead?

Reply:
I think the method is pretty harmless.
I like the the idea of using simplifying the code with a set -- one could do something like
```python
node_configs = [node_type["node_config"] for node_type in config["available_node_types"].values()]
return len({node_config.get(key) for node_config in node_configs}) == 1
```

Reply:
I like the set method as well and will go with that if there aren't any objections. We can consider dropping the extra logging logic later if we think it's adding too much complexity

Reply:
Yep, this is what I had in mind @DmitriGekhtman . I am not sure if we should also assert that the value in the set is not None.

Reply:
Going with set approach. Also noticed that this same_everywhere is only ever used once inside of print_info, so merged the logic into print_info so that its a bit less confusing to read going forward